### PR TITLE
Replace Bluebird with native promise implementation

### PR DESCRIPTION
**Pull request by andreogle**
### Description

We only use Bluebird for promise timeouts. After some investigation and comments by @Siegrift it seemed like it could be replaced with native JS promises (using `Promise.race`). The initial comments are no longer applicable

### CODE REVIEW

File: packages/node/__dev__/invoke.ts [hunk 1]
```diff
@@ -7,4 +7,6 @@ async function invoke() {
   await local.startCoordinator();
 }
 
-invoke();
+invoke()
+  .then(() => process.exit(0))
```
Comment (Siegrift) [inline packages/node/__dev__/invoke.ts:11]: @andreogle Maybe add a comment why this is needed

File: packages/node/src/utils/promise-utils.ts [hunk 2]
```diff
@@ -25,22 +24,15 @@ export function goTimeout<T>(ms: number, fn: Promise<T>): Promise<Response<T>> {
...
-//
-// The problem with this is that that the slow promise still runs until it resolves.
-// This means that the serverless function will not exit until the entire timeout
-// duration has been reached which is a problem.
 export function promiseTimeout<T>(ms: number, promise: Promise<T>): Promise<T> {
-  return Bluebird.resolve(promise).timeout(ms);
+  const timeout = new Promise((_res, reject) => {
+    const id = setTimeout(() => {
+      clearTimeout(id);
+      reject(new Error(`operation timed out in ${ms} ms.`));
```
Comment (Siegrift) [inline packages/node/src/utils/promise-utils.ts:31]:
```suggestion
      reject(new Error(Operation timed out in ${ms} ms.`));
```

File: packages/node/src/utils/promise-utils.ts [hunk 3]
```diff
@@ -25,22 +24,15 @@ export function goTimeout<T>(ms: number, fn: Promise<T>): Promise<Response<T>> {
...
-//   }
-//
-// The problem with this is that that the slow promise still runs until it resolves.
-// This means that the serverless function will not exit until the entire timeout
-// duration has been reached which is a problem.
 export function promiseTimeout<T>(ms: number, promise: Promise<T>): Promise<T> {
-  return Bluebird.resolve(promise).timeout(ms);
+  const timeout = new Promise((_res, reject) => {
+    const id = setTimeout(() => {
+      clearTimeout(id);
```
Comment (Siegrift) [inline packages/node/src/utils/promise-utils.ts:30]: `clearTimeout` is useless here.

You want to call it other way around -- when the original promise resolves/rejects.
(Maybe that's the reason you need the process.exit calls?)

**Review by Siegrift — CHANGES_REQUESTED** [review]
You forgot to remove bluebird from one of the test files.

File: packages/node/src/utils/promise-utils.ts (hunk 3 above)
Comment (andreogle) [reply packages/node/src/utils/promise-utils.ts:30]: Thanks, how about this?

```ts
let timeoutId: NodeJS.Timeout;
const timeout = new Promise((_res, reject) => {
  timeoutId = setTimeout(() => {
    reject(new Error(`Operation timed out in ${ms} ms.`));
  }, ms);
});

const wrappedPromise = promise.finally(() => {
  if (timeoutId) {
    clearTimeout(timeoutId);
  }
});

return Promise.race([wrappedPromise, timeout]) as Promise<T>;
```

Comment (Siegrift) [reply packages/node/src/utils/promise-utils.ts:30]: Yep, lgtm. Did it fix the issue with explicit process.exit call?

**Review by Siegrift — APPROVED** [review]

File: packages/node/src/utils/promise-utils.ts (hunk 3 above)
Comment (andreogle) [reply packages/node/src/utils/promise-utils.ts:30]: It did actually fix it 👍🏻

**Pull request merged**